app/src/pages/common/settingsPage/patternAnalysisTab/patternListHeader/patternListHeader.jsx

```diff
@@ -25,28 +27,44 @@ export const messages = defineMessages({
 });
 
 @injectIntl
+@connect(
+  (state) => ({
+    PAState: PAStateSelector(state),
+  }),
+  {
+    updatePAState: updatePAStateAction,
+  },
+)
 export class PatternListHeader extends Component {
   static propTypes = {
     intl: intlShape.isRequired,
     onAddPattern: PropTypes.func,
+    PAState: PropTypes.bool.isRequired,
+    updatePAState: PropTypes.func,
   };
 
   static defaultProps = {
     pattern: {},
     id: 0,
     onAddPattern: () => {},
+    updatePAState: () => {},
+  };
+
+  onToggleEnablePA = (value) => {
+    this.props.updatePAState(value);
   };
 
   render() {
     const {
       intl: { formatMessage },
       onAddPattern,
+      PAState,
     } = this.props;
     return (
       <div className={cx('pattern-list-header')}>
         <span className={cx('caption')}>{formatMessage(messages.patternAnalysis)}</span>
         <span className={cx('switcher')}>
-          <InputBigSwitcher />
+          <InputBigSwitcher mobileDisabled value={PAState} onChange={this.onToggleEnablePA} />
```

Comment:
May be just `this.props.updatePAState`?)
Seems that `this.onToggleEnablePA` function is redundant

Reply:
Fixed.